luigi_pipeline/lib/model/base_mt_schema.py

```diff
@@ -109,30 +92,104 @@ def c_1(self):
 
     """
     def __init__(self, mt):
-        self.mt = mt
+        self._mt = None
+        self.set_mt(mt)
         self.mt_instance_meta = {
             'row_annotations': defaultdict(lambda: {
                 'annotated': 0,
                 'result': {},
             })
         }
 
+    @property
+    def mt(self):
+        """
+        Don't allow direct sets to self.mt to ensure some references are updated
+        """
+        return self._mt
+
+    # Don't use @mt.setter as it makes inheritance harder
+    def set_mt(self, mt):
+        """Set mt here"""
+        self._mt = mt
+
     def all_annotation_fns(self):
         """
         Get all row_annotation decorated methods using introspection.
         :return: list of all annotation functions
         """
-        return getmembers(self, lambda x: ismethod(x) and hasattr(x, 'mt_cls_meta'))
+        return [a[1] for a in getmembers(self, lambda x: isinstance(x, RowAnnotation))]
 
     def annotate_all(self, overwrite=False):
         """
         Iterate over all annotation functions and call them on the instance.
         :return: instance object
         """
-        for atn_fn in self.all_annotation_fns():
-            getattr(self, atn_fn[0])(overwrite=overwrite)
+        called_annotations = set()
+        rounds: List[List[RowAnnotation]] = [self.all_annotation_fns()]
+        print(f'Will attempt to apply {len(rounds[0])} row annotations')
```

Comment:
we use `loggger`, not `print` for logging messages. Anywhere that used `print` for debugging reasons can probably taken out, and anywhere that had useful information for production can get switched to `logger.info`